runtime/src/main/cpp/Memory.cpp

```diff
@@ -838,6 +820,11 @@ void ObjHeader::destroyMetaObject(TypeInfo** location) {
     WeakReferenceCounterClear(meta->counter_);
     UpdateRef(&meta->counter_, nullptr);
   }
+
+#ifdef KONAN_OBJC_INTEROP
+  Kotlin_ObjCExport_releaseAssociatedObject(meta->associatedObject);
```

Comment:
By convention we add `_` to the end of C++ field names, i.e. `associatedObject_`